tests/e2e/kubetest2-kops/util/publicip.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"time"
+)
+
+const externalIPMetadataURL = "http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip"
+
+var externalIPServiceURLs = []string{
+	"https://ip.jsb.workers.dev",
+	"https://v4.ifconfig.co",
+}
+
+// ExternalIPRange returns the CIDR block for the public IP
+// in front of the kubetest2 client
+func ExternalIPRange() (string, error) {
```

Comment:
This is mostly copied from [kubetest/kops.go](https://github.com/kubernetes/test-infra/blob/1844d8648b668b8b95c3efdbc54f914db0e3c742/kubetest/kops.go#L505-L539) along with http.go above